backend/app/services/tables/base.py

```
@@ -0,0 +1,115 @@
+            user_name: Username
+
+        Returns:
+            True if user has access, False otherwise
+        """
+        pass
+
+
+class TableProviderRegistry:
+    """Provider registry."""
+
+    _providers: Dict[str, Type[BaseTableProvider]] = {}
```

Ruff (0.14.10)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)